packages/components/organisms/f-form/src/components/Form.vue

```diff
@@ -25,50 +47,71 @@
 
 
 <script>
+import { VueGlobalisationMixin } from '@justeat/f-globalisation';
 import FButton from '@justeat/f-button';
 import FormField from '@justeat/f-form-field';
-import { globalisationServices } from '@justeat/f-services';
+import ErrorMessage from '@justeat/f-error-message';
+import FormValidationMixin from '../mixin/formValidation.mixin';
 import tenantConfigs from '../tenants';
 import { DEFAULT_BUTTON_TEXT, FORM_EVENTS, PROP_VALIDATION_MESSAGES } from '../constants';
+import { FormFieldClass, FormFieldErrorClass } from '../services/formField';
 
 export default {
     components: {
         FButton,
+        ErrorMessage,
         FormField
     },
 
+    mixins: [
+        FormValidationMixin,
+        VueGlobalisationMixin
+    ],
+
     props: {
         formData: {
             type: Object,
             required: true
         },
 
+        locale: {
+            type: String,
+            default: 'en-GB'
+        },
+
         isFormSubmitting: {
             type: Boolean,
             default: false
         }
     },
 
     data () {
-        const locale = globalisationServices.getLocale(tenantConfigs, this.locale, this.$i18n);
-        const localeConfig = tenantConfigs[locale];
-
         return {
-            copy: { ...localeConfig }
+            tenantConfigs
         };
     },
 
     computed: {
         formFields () {
             const formFields = {};
 
-            this.formData.formFields.forEach(field => {
-                formFields[field.name] = field.value;
-            });
+            if (this.formData.formFields?.length) {
+                this.formData.formFields.forEach(field => {
+                    formFields[field.name] = field.value;
+                });
+            }
 
             return formFields;
         },
 
+        formFieldData () {
+            if (this.formData.formFields?.length) {
+                return this.formData.formFields.map(field => this.createField(field));
+            }
+
+            return [];
```

Comment:
`map` actually handles empty arrays already

```javascript
[].map(x => x);
// []
```

But it is still good practice to do some type-checking:
```suggestion
            if (Array.isArray(this.formData.formFields)) {
                return this.formData.formFields.map(field => this.createField(field));
            }

            return [];
```